Updates query DSL index page (#2451)
opensearch-project/documentation-website: _opensearch/query-dsl/term.md

layout: default
title: Term-level queries
parent: Query DSL
nav_order: 65
---

# Term-level queries

Term-level queries search an index for all documents that contain a search term. Documents returned by a term-level query are not sorted by their relevance scores.

When working with text data, use term-level queries for fields mapped as keyword only.

Term-level queries are not suited for searching analyzed text fields. To return analyzed fields, use a full-text query.

## Term-level query types

You can use any of the term-level query types shown in the following table:

| Query type | Description
:--- | :--- | :---

> `term` | Specifies a search for an exact term for a provided field.

Reviewer: "Specifies a search for an exact term in the field provided." ?

`terms` | Specifies a search for multiple terms in the same provided field.
`ids` | Specifies a search for one or more documents by the ID value.
`range` | Specifies a search within an upper and lower bound of values in a provided field.
`prefix` | Specifies a search for terms that begin with a prefix.
`exists` | Specifies a search for documents that contain a specific field.
`wildcard` | Specifies a search for terms that match a wildcard pattern. Indicate matching zero or more characters with an asterisk (*) or a single character by a question mark (?).

> `regex` | Specifies a search for terms that match a regular expression.

Reviewer 2: ```suggestion
`regexp` | Specifies a search for terms that match a regular expression.
```

> 

Reviewer 2: I haven't tested the queries above, but I think we also support `fuzzy` and `terms set` queries.

Author: sounds good. verified terms_set definition for term-level queries. Fuzzy queries are a sub-type of full-text queries.

Author: ok. i'm researching the definition for fuzzy term-level queries now. we don't have it defined yet. Thanks!

> `terms_set` | Similar to a `terms` query, but you specify the number of matching terms that are required to return a document.

Reviewer 2: Please add the `fuzzy` query in this list. It is a subtype of term-level queries.

Reviewer 3: @alicejw-aws , @macohen is taking another look at this from a technical standpoint since we want to make sure it is 100% technically correct before publishing it.

Author: @hdhalter thanks, I appreciate that.

# Term-level queries

Term-level queries search an index for all documents that contain a search term. Documents returned by a term-level query are not sorted by their relevance scores.

When working with text data, use term-level queries for fields mapped as keyword only.

Term-level queries are not suited for searching analyzed text fields. To return analyzed fields, use a full-text query.

## Term-level query types

You can use any of the term-level query types shown in the following table:

| Query type | Description
:--- | :--- | :---
`term` | Specifies a search for an exact term for a provided field.
`terms` | Specifies a search for multiple terms in the same provided field.
`ids` | Specifies a search for one or more documents by the ID value.
`range` | Specifies a search within an upper and lower bound of values in a provided field.
`prefix` | Specifies a search for terms that begin with a prefix.
`exists` | Specifies a search for documents that contain a specific field.

> `wildcard` | Specifies a search for terms that match a wildcard pattern. Indicate all values with an asterisk (*) or a single value by a question mark (?).

Reviewer 2: ```suggestion
`wildcard` | Specifies a search for terms that match a wildcard pattern. Indicate matching zero or more characters with an asterisk (*) or a single character by a question mark (?).
```

`range` | Specifies a search within an upper and lower bound of values in a provided field.
`prefix` | Specifies a search for terms that begin with a prefix.
`exists` | Specifies a search for documents that contain a specific field.

> `fuzzy` | Specifies a [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance) that measures the number of one-character changes needed to change one term to another term. The fuzzy query creates sets of all possible variations of a term specified by the `value` field, and returns exact matches for terms that match the specified edit distance.    

Reviewer 2: I don't think we should go into details here because we're not doing it for other query types.
```suggestion
`fuzzy` | Returns terms that are similar to the search term within the maximum allowed [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance). The Levenshtein distance measures the number of one-character changes needed to change one term to another term.
```

Author: ok thanks for fine-tuning the definition.

}
```

> The response indicates zero `hits` to denote a match. This is because the term “To be, or not to be” is searched literally in the inverted index, where only the analyzed values of the text fields are stored.

Reviewer 2: ```suggestion
The response contains no matches, indicated by zero `hits`. This is because the term “To be, or not to be” is searched literally in the inverted index, where only the analyzed values of the text fields are stored.
```
